spring-cloud-gcp-starters/spring-cloud-gcp-starter-config/src/main/java/org/springframework/cloud/gcp/config/autoconfig/GcpConfigAutoConfiguration.java

```diff
@@ -34,5 +37,12 @@
 
 	@Autowired
 	GcpConfigProperties gcpConfigProperties;
+
+	// Disabling the default Spring cloud config property source locator
+	@Bean
```

Comment:
@joaoandremartins @spencergibb @saturnism 
Please see if this is an acceptable solution to disable the config property source locator.

Reply:
I'm confused at why you import spring-cloud-starter-config at all. I don't think you need it. Bootstrap comes from spring-cloud-commons which you could get with spring-cloud-starter

Reply:
We want to reuse the same property namespace as `ConfigClientProperties`. We could import `spring-cloud-config-client` instead of the whole `spring-cloud-config-starter`, but the same problem would pose -- `ConfigServicePropertySourceLocator` would also be imported from the same package and it will try to run unless we tell it not to (and we don't want our users to do it through properties).

Basically, if our starter is active, we want to default `ConfigClientProperties.enabled` to false. Is this a good way to do so?

Reply:
Reusing the same namespace seems like a bad idea.

Reply:
@spencergibb : If that is the case, I can simplify what's done here easily.